ee/errgroup/errgroup.go

```diff
@@ -0,0 +1,163 @@
+package errgroup
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"time"
+
+	"github.com/pkg/errors"
+	"golang.org/x/sync/errgroup"
+)
+
+type LoggedErrgroup struct {
+	errgroup *errgroup.Group
+	cancel   context.CancelFunc
+	doneCtx  context.Context // nolint:containedctx
```

Comment:
This is the same nolint exception we had previously, just moved from osqueryinstance to here. In the future I think we might be able to get rid of this entirely -- but the errgroup is very sensitive when it comes to updating anything that touches `doneCtx` so I want to do it in a separate PR at a later time.